Odata-docs/webapi-8/fundamentals/query-options.md

```diff
@@ -0,0 +1,486 @@
+---
+title: "OData query Options"
+description: "This documentation describes how to use the supported query options."
+
+author: KenitoInc
+ms.author: kemunga
+
+ms.date: 12/16/2022
+---
+# OData query options
+**Applies To**:[!INCLUDE[appliesto-webapi](../../includes/appliesto-webapi-v8.md)]
+
+AspNetCore OData 8.x provides out of the box querying capabilities.
+
+A query option is basically requesting that a service perform a set of transformations such as filtering, sorting, etc. to its data before returning the results. See more information [here](/odata/concepts/queryoptions-overview).
+
+## Model class
+We will define our model as follows:
+
+```csharp
+public class Customer
+{
+    [Key]
+    public int Id { get; set; }
+    public String Name { get; set; }
+    public int Age { get; set; }
+    public List<Order> Orders { get; set; }
+}
+
+public class Order
+{
+    [Key]
+    public int Id { get; set; }
+    public int Price { get; set; }
+    public int Quantity { get; set; }
+}
+```
+
+We have a `Customer` entity with `Orders` navigation property.
+
+## Configuring Startup class
+```csharp
+public class Startup
+{
+    public void ConfigureServices(IServiceCollection services)
+    {
+        var modelBuilder = new ODataConventionModelBuilder();
+        modelBuilder.EntitySet<Customer>("Customers");
+
+        services.AddControllers().AddOData(
+            options => options.Select().Filter().OrderBy().Expand().Count().SetMaxTop(null).AddRouteComponents(
+                "odata",
+                modelBuilder.GetEdmModel()));
+    }
+}
+```
+
+In the above configuration, `options` is an instance of `ODataOptions`.
+`Select()` enables `$select` query option.
+`Filter()` enables `$filter` query option.
+`OrderBy()` enables `$orderby` query option.
+`Expand()` enables `$expand` query option.
+`Count()` enables `$count` query option.
+`SetMaxTop()` sets the maximum value of `$top` that a client can request.
+
+## Controller
+
+```csharp
+public class CustomersController : ODataController
+{
+    [EnableQuery]
+    public IActionResult Get()
+    {
+        return Ok(context.Customers);
```

Comment:
I don't know if you wanted to include code snippets that can actually compile but this definitely wouldn't because `context` is undefined. Something that would take very little effort to do is define a static `List<Customer>` field in your controller and populate it with say 3 customer objects and use it as your data source. That way the code would compile if someone who's going through the doc decided to try-it-out.

Reply:
Fixed